src/mobu/jupyterclient.py

```diff
@@ -295,17 +296,21 @@ async def is_lab_stopped(self, final: bool = False) -> bool:
         return result
 
     @_convert_exception
-    async def spawn_lab(self) -> JupyterImage:
+    async def spawn_lab(self) -> ControllerImage:
         spawn_url = self.jupyter_url + "hub/spawn"
 
         # Determine what image to spawn.
         if self.config.image_class == JupyterImageClass.RECOMMENDED:
-            image = await self.cachemachine.get_recommended()
+            image = await self.controller.get_recommended()
         elif self.config.image_class == JupyterImageClass.LATEST_WEEKLY:
-            image = await self.cachemachine.get_latest_weekly()
+            image = await self.controller.get_latest_weekly()
         else:
             assert self.config.image_reference
-            image = JupyterImage.from_reference(self.config.image_reference)
+            ref = self.config.image_reference
+            image = ControllerImage(
+                path=ref, name=ref, digest="unknown", tags={}
```

Comment:
`digest` and `tags` should be optional in the model so that you can just omit them here.